### PR TITLE
Kill CProfilesManager singleton

### PR DESCRIPTION
**Pull request by garbear**
This kills the CProfilesManager singleton by moving it to ServiceManager. Of course, everything depends on CProfilesManager, including a multitude of other globals/singletons, so I started moving some of these to ServiceManager as well.

## Motivation and Context
Make solving the bug tackled by #13412 more tractable.

## How Has This Been Tested?
CPlayerCoreFactory - tested playing a video
CDatabaseManager - tested starting with an existing userdata and a clean userdata
CEventLog - tested by viewing Event Log
CProfilesManager - tested by creating and using several profiles

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by garbear** [comment]
jenkins build this please

**Comment by jimfcarroll** [comment]
Hey garbear. Long time. If you can, pop into the team channel when you have a chance. I want to talk to you about where the ServiceManager is going.

File: xbmc/input/IRTranslator.h [hunk 1]
```diff
@@ -49,6 +50,9 @@ class CIRTranslator
   bool LoadIRMap(const std::string &irMapPath);
   void MapRemote(TiXmlNode *pRemote, const std::string &szDevice);
 
+  // Construction parameters
+  const CProfilesManager &m_profileManager;
```
Comment (Rechi) [inline xbmc/input/IRTranslator.h]: unused if `!(defined(HAS_LIRC) || defined(HAS_IRSERVERSUITE))`

Comment (garbear) [reply xbmc/input/IRTranslator.h]: "fixed" (still logically unused at runtime, but at least the compiler will see it)

**Comment by garbear** [comment]
All green. Rebased and squashed. jenkins build and merge

**Pull request merged**